src/platform/android/DnssdImpl.cpp

```diff
@@ -183,6 +183,20 @@ CHIP_ERROR ChipDnssdStopBrowse(intptr_t browseIdentifier)
     return CHIP_ERROR_NOT_IMPLEMENTED;
 }
 
+CHIP_ERROR extractProtocol(const char * serviceType, char * outServiceName, DnssdServiceProtocol & outProtocol)
+{
+    std::string serviceTypeStr(serviceType);
+    size_t index = serviceTypeStr.find(".");
+
+    ReturnErrorCodeIf(index == std::string::npos, CHIP_ERROR_INVALID_ARGUMENT);
+
+    CopyString(outServiceName, index + 1, serviceType);
```

Comment:
outServiceName needs to be sized - either use a span or add a length argument to ensure we do not overflow.

You should not pass index+1 here but rather `outServiceNameLength`

Reply:
Thanks your kindly reviewing. I added.